packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -130,18 +132,20 @@ const SEQUENCER_MNEMONIC = env.SEQUENCER_MNEMONIC || env.MNEMONIC
 const PROPOSER_MNEMONIC = env.PROPOSER_MNEMONIC || env.MNEMONIC
 const SEQUENCER_HD_PATH = env.SEQUENCER_HD_PATH || env.HD_PATH
 const PROPOSER_HD_PATH = env.PROPOSER_HD_PATH || env.HD_PATH
-const DEBUG_IMPERSONATE_SEQUENCER_ADDRESS =
-  env.DEBUG_IMPERSONATE_SEQUENCER_ADDRESS
-const DEBUG_IMPERSONATE_PROPOSER_ADDRESS =
-  env.DEBUG_IMPERSONATE_PROPOSER_ADDRESS
-if (
-  (DEBUG_IMPERSONATE_SEQUENCER_ADDRESS &&
-    !DEBUG_IMPERSONATE_PROPOSER_ADDRESS) ||
-  (!DEBUG_IMPERSONATE_SEQUENCER_ADDRESS && DEBUG_IMPERSONATE_PROPOSER_ADDRESS)
-) {
-  throw new Error(
-    'Both sequencer & proposer addresses must be impersonated or none at all.'
-  )
+const BLOCK_OFFSET = parseInt(env.BLOCK_OFFSET, 10) || 1 // Default is 1 because Geth normally has 1 more block than L1
+let DEBUG_IMPERSONATE_SEQUENCER_ADDRESS
+let DEBUG_IMPERSONATE_PROPOSER_ADDRESS
+if (env.USE_HARDHAT) {
+  DEBUG_IMPERSONATE_SEQUENCER_ADDRESS = env.DEBUG_IMPERSONATE_SEQUENCER_ADDRESS
+  DEBUG_IMPERSONATE_PROPOSER_ADDRESS = env.DEBUG_IMPERSONATE_PROPOSER_ADDRESS
+  if (
+    !DEBUG_IMPERSONATE_SEQUENCER_ADDRESS ||
```

Comment:
This needs to be `!a && !b`

Reply:
right now it says if either is set then throw an error. I think we should just enforce setting both of them